galley/pkg/config/analysis/analyzers/annotations/annotations.go

```diff
@@ -111,7 +52,7 @@ func (*K8sAnalyzer) Metadata() analysis.Metadata {
 // Analyze implements analysis.Analyzer
 func (fa *K8sAnalyzer) Analyze(ctx analysis.Context) {
 	ctx.ForEach(metadata.K8SCoreV1Endpoints, func(r *resource.Entry) bool {
-		fa.allowAnnotations(r, ctx, "Endpoints", metadata.K8SCoreV1Endpoints)
+		fa.allowAnnotations(r, ctx, "Endpoint", metadata.K8SCoreV1Endpoints)
```

Comment:
Is this a bugfix for something? If so, do we have test coverage for it?

I'm nervous about tweaking this unless we're sure we know what it does. :) @esnible can you comment here?

Reply:
It is just a naming change so that we are not using plural for the resource name.

@esnible I am not sure why we need to check `Endpoint` though.

Reply:
The resource name **is plural**, though.  `kubectl get endpoints details -o yaml` gives

```
apiVersion: v1
kind: Endpoints
metadata:
...
```

(Unlike everything else in Kubernetes, `Endpoints` is not the plural of `Endpoint`.  There is no `Endpoint`.)

We are checking `Endpoints` because no Istio annotations should be applied to them.  We already have them in memory, so it shouldn't cost much to iterate through them and make sure no fool tried to put an Istio annotation upon an endpoint.

Reply:
Done. Added a test for `Endpoints`